### PR TITLE
Convert colour.js to typescript

### PR DESCRIPTION
**Pull request by MarkusJx**
This will convert ``static/colour.js`` to typescript (see #2981)

Changes are pretty much the same as with the other conversions I did.

Assumed that param ``colours`` in ``applyColours`` is of type ``number[]`` as this is the only fitting type.

### CODE REVIEW

**Review by supergrecko — APPROVED** [review]
Looks good, one tiny nit

File: static/colour.ts [hunk 1]
```diff
@@ -24,11 +24,21 @@
 
 'use strict';
```
Comment (supergrecko) [inline static/colour.ts:25]: Unnecessary 'use strict'

**Comment by MarkusJx** [comment]
Done.

**Comment by supergrecko** [comment]
Thank you so much for your contribution!

**Pull request merged**